samplr/README.md

```diff
@@ -51,3 +51,162 @@ e.g.
 ```bash
 samplrctl snippets list /tmp/local-repository
 ```
+
+## Testing
+
+To test the samplr code `cd` into the samplr directory and run
+
+```bash
+go test -v -race ./...
+```
+
+This will run the tests in the `samplr` directory as well as all subdirectories in it, recursively.
+
+### Integration Tests
+
+The integration tests take a bit longer to run, and require network access, so they require a special
+command line option to run: `-tags integration`
+
+```bash
+go test -v -tags integration ./...
+```
+
+This uses the Go `tag` feature, which is enabled by putting a comment in a file of form
+
+```go
+// +build TAGNAME
+```
+
+## Deploying
+
+The `Makefile` has several variables that can be overridden via environment variables
+in order to customize how the deployment occurs.
+
+```makefile
+# The (gcloud) test cluster that is being worked against
+GCP_CLUSTER_NAME ?= devrel-services
```

Comment:
these vars should default to the "DEV" value so that deploying to DEV is just `make deploy`. There's currently a mismatch of which vars need to be overridden depending on which environment one is deploying to.